### ISSUE TITLE
[BUG]: GroupTimeSeriesSplit CV Splitter not working as expected in PyCaret Classification

### ISSUE THREAD

**Issue by diarmaidfinnerty** [pycaret/pycaret#3348]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I would like to use pycaret to train a classification model on data that contains some time dependent variables. I want to split the dataset into training and test sets with a gap of 3 month (1 quarter) between the training and the testing set - Training data will be Q1 & Q2 of the year, Q3 excluded as a gap, and Q4 being the test set for this split. 

Eventually I would like to expand this to a much larger data covering multiple years with rolling training data, gaps and test data like below. 

- Split 1 - Train Q1 & Q2 of Year 1 - Test Q4 of Year 1
- Split 2 - Train Q2 & Q3 of Year 1 - Test Q1 of Year 2
- Split 3 - Train Q3 & Q4 of Year 1 - Test Q2 of Year 2

....and so on. 

`mlextend` provides an excellent method to do this using `GroupTimeSeriesSplit` where the group in this case is the column `quarter`. Here it is a graph with a toy example of a single split. It clearly is working as intended 

<img width="1086" alt="Screenshot 2023-02-28 at 15 26 59" src="https://user-images.githubusercontent.com/15366181/221899388-cd8c1ee4-72f4-4f6a-a1dd-e7211d77bc35.png">

However, when this cv splitter is passed into PyCaret, it seems that it gets completely ignored and the data is split as it normally would be.

Can you advise? Is there an issue with my implementation 

### Reproducible Example

```python
import pandas as pd
import numpy as np

from mlxtend.evaluate.time_series import (
    GroupTimeSeriesSplit,
    plot_splits,
    print_cv_info,
    print_split_info,
)

# Unfortunately I can't share the data. 
# Only important that there is a date feature that spans the entire year
data = pd.read_csv('./somerandomdata')
data['quarter'] = data.date.apply(lambda x: pd.Timestamp(x).quarter)

X = data
groups = data.quarter
y = np.random.randint(2, size=data.shape[0])

# demonstrates what the GroupTimeSeriesObject does - should be identical to image provided
cv_args = {"test_size": 1, "gap_size": 1, "n_splits" : 1}
plot_splits(X, y, groups, **cv_args)
print_split_info(X, y, groups, **cv_args)

# Training here
from pycaret.classification import *

train = data.copy()
train['target'] = y
train['group'] = groups
cv = GroupTimeSeriesSplit(**cv_args)

pyc = setup(data=train.drop('date', axis=1),
       target='target',
       data_split_shuffle=False,
       data_split_stratify=False,
       fold_groups='group',
       fold_strategy=cv,
       preprocess=False)

best = pyc.compare_models(errors='raise')

# Show data splits 
train_qs = pyc.get_config('X_train_transformed')['quarter'].unique()
print(f'Quarter num in training data {train_qs}')
test_qs = pyc.get_config('X_test_transformed')['quarter'].unique()
print(f'Quarter num in testing data {test_qs}')
```

### Expected Behavior

`>> Quarter num in training data [1 2]`
`>> Quarter num in testing data [4]`

### Actual Results

```python-traceback
`>> Quarter num in training data [1 2 3]`
`>> Quarter num in testing data [3 4]`
```

### Installed Versions

<details>
Replace this line with the output of the version code above.
</details>'
python==3.8.15

pycaret==3.0.0rc8
mlxtend==0.21.0

**Comment by diarmaidfinnerty** [comment]
Hi, has there been any investigation into this? Is there more information that I could provide to move this along?

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
